client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -415,6 +415,29 @@ function mediaButton( editor ) {
 		}
 	} );
 
+	editor.addButton( 'wp_img_edit', {
+		tooltip: i18n.translate( 'Edit', { context: 'verb' } ),
+		icon: 'dashicon dashicons-edit',
+		onclick: function() {
+			const selectedSite = getSelectedSiteFromState();
+			const node = editor.selection.getNode();
+			const m = node.className.match( /wp-image-(\d+)/ );
+			const imageId = m && parseInt( m[ 1 ], 10 );
+			const image = MediaStore.get( selectedSite.ID, imageId );
+
+			// Cause edit modal to show...
+			if ( ! selectedSite ) {
```

Comment:
Let's move this statement all the way to the top after
```
const selectedSite = getSelectedSiteFromState();
```

As it this may throw an error on, when selectedSite is `null`:
```
const image = MediaStore.get( selectedSite.ID, imageId );
```

I also prefer using a local variable over calling `selectedSite.ID` several times.
```
const siteId = selectedSite.ID;
```

The comment is also no longer relevant